File: map/MapUpdateFunctional.py
scores = {
    "Alice": 90,
    "Bob": 80,
    "Charlie": 85
}

# 条件に一致するキーの値を更新した新しい辞書を作成する
updated_scores = {key: value + 5 if key ==
                  "Bob" else value for key, value in scores.items()}

print("Updated scores:", updated_scores)

File: map/MapUpdateFunctional.ts
let scores: Map<string, number> = new Map([
  ["Alice", 90],
  ["Bob", 80],
  ["Charlie", 85],
]);

// 条件に一致するキーの値を更新した新しいMapを作成する
let updatedScores: Map<string, number> = new Map(
  Array.from(scores.entries()).map(([key, value]) => {
    return [key, key === "Bob" ? value + 5 : value];
  })
);

console.log("Updated scores:", updatedScores);
